server/services/RuleService.ts

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {
+  OpenSearchDashboardsRequest,
+  OpenSearchDashboardsResponseFactory,
+  IOpenSearchDashboardsResponse,
+  ResponseError,
+  RequestHandlerContext,
+  ILegacyCustomClusterClient,
+} from 'opensearch-dashboards/server';
+import {
+  CreateRuleParams,
+  CreateRuleResponse,
+  DeleteRuleParams,
+  DeleteRuleResponse,
+  GetRulesParams,
+  GetRulesResponse,
+  UpdateRuleParams,
+  UpdateRuleResponse,
+} from '../models/interfaces';
+import { CLIENT_RULE_METHODS } from '../utils/constants';
+import { Rule } from '../../models/interfaces';
+import { ServerResponse } from '../models/types';
+import { safeDump } from 'js-yaml';
+import moment from 'moment';
+
+export default class RulesService {
+  osDriver: ILegacyCustomClusterClient;
+
+  constructor(osDriver: ILegacyCustomClusterClient) {
+    this.osDriver = osDriver;
+  }
+
+  /**
+   * Calls backend POST Rules API.
+   */
+
+  createRule = async (
+    _context: RequestHandlerContext,
+    request: OpenSearchDashboardsRequest,
+    response: OpenSearchDashboardsResponseFactory
+  ): Promise<IOpenSearchDashboardsResponse<ServerResponse<CreateRuleResponse> | ResponseError>> => {
+    try {
+      const {
+        id,
+        title,
+        description,
+        detection,
+        status,
+        author,
+        references,
+        tags,
+        log_source,
+        level,
+        false_positives,
+      } = request.body as Rule;
+      const today = moment(moment.now()).format('YYYY/MM/DD');
+      const jsonPayload = {
+        id,
+        title,
+        description,
+        status,
+        author,
+        date: today,
+        modified: today,
+        references: references.map((ref) => ref.value),
+        tags: tags.map((tag) => tag.value),
+        logsource: {
+          product: log_source,
+          service: log_source,
+        },
+        level,
+        falsepositives: false_positives.map((falsePos) => falsePos.value),
+        detection: JSON.parse(detection),
+      };
+      console.log(jsonPayload);
```

Comment:
Not blocking for this draft, but are these development logs that can be removed, or are we planning to leave this in?

Reply:
Will be removed